api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto

```diff
@@ -183,44 +183,162 @@ message JwtHeader {
   string value_prefix = 2;
 }
 
-// This is the Envoy HTTP filter config for JWT authentication.
-// [#not-implemented-hide:]
-message JwtAuthentication {
-  // List of JWT rules to valide.
-  repeated JwtRule rules = 1;
+// This message specifies a Jwt requirement.
+message RequiredProvider {
+  // Specify the issuer for the required JwtProvider.
```

Comment:
This uniquely identifier a single JwtProvider already specified elsewhere?

Reply:
I don't believe the current proposal has a unique identifier, it's using the issuer as the id and allowing overrides on audiences.

If we model this closer to the googleapis model, we'd add a user-specified id which allows for multiple provider configurations with the same issuer (e.g. with different audience sets). This also addresses the other comment about audience overrides. I'm in favor of this, but don't feel very strongly about it.

Reply:
Yes, "issuer" uniquely identifies the Provider specified in the "providers" field. 

Hi @llchan based on my understanding of googleapis AuthRequirement,  it is similar to what I have proposed here. Here is its example.

-- selector:  /foo
   requirements:
      provider_id: issuer1
      audiences: [audience1, audience2]

Reply:
The google api I'm referring to is [this one](https://github.com/googleapis/googleapis/blob/master/google/api/auth.proto#L79). Notice that the "primary key" is not the issuer, but rather a user-defined id. This allows multiple configurations to be defined per issuer, so the routes can refer to those specific configurations rather than overriding a single master config for that issuer.

I think it's fairly unlikely anyone needs really really fine grained per-route audience overrides. More likely, there's some big giant issuer that can issue tokens for multiple audiences, and routes require a particular issuer+audience pair. By letting the user id-ify these pairs we allow route requirements to refer to shared configurations.

Reply:
Also in case it wasnt clear, id-ifying these obviates the need for the audience overrides, which I think makes the whole thing a little more clear.